### PR TITLE
Fix server

### PR DESCRIPTION
**Pull request by erogol**
- use given arguments, if they are defined in server.py 
- bug fix regarding gradual training

### CODE REVIEW

File: server/server.py [hunk 1]
```diff
@@ -30,14 +30,14 @@ def convert_boolean(x):
 embedded_model_folder = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'model')
 checkpoint_file = os.path.join(embedded_model_folder, 'checkpoint.pth.tar')
 config_file = os.path.join(embedded_model_folder, 'config.json')
+config = create_argparser().parse_args()
 
-if os.path.isfile(checkpoint_file) and os.path.isfile(config_file):
+if config.tts_checkpoint is None and config.tts_config is None:
+    print(f" >  Loading default server model from {embedded_model_folder}.")
```
[the remark this answers is not in the document]
Comment (erogol) [reply server/server.py]: but we already set python >= 3.6 as the requirement.

[the remark this answers is not in the document]
Comment (reuben) [reply server/server.py]: Then ignore this comment :)

**Comment by erogol** [comment]
> 
> 
> I don't understand the `server.py` changes. What is broken?

it did not use the given arguments if there are checkpoint and config file under ```embedded_model_folder```. I guess it is more reasoanble to use the default model if there are no given tts_checkpoint and tts_config.

**Comment by reuben** [comment]
It's supposed to only use it if they are not given. How did you test it?

The problem with these changes is that they cause command line arguments to be parsed when you're just importing the file. That should only be done inside the `__main__` block when the script is being called directly.

**Comment by reuben** [comment]
Ah, I see what's happening, it only parses the CLI flags if the synthesizer or config haven't been set.

**Comment by reuben** [comment]
Here's a fix that avoid parsing arguments when imported. Can you test this?

```diff
diff --git a/server/server.py b/server/server.py
index 2831c75..d40e242 100644
--- a/server/server.py
+++ b/server/server.py
@@ -55,8 +55,11 @@ def tts():
 
 
 if __name__ == '__main__':
-    if not config or not synthesizer:
-        args = create_argparser().parse_args()
+    args = create_argparser().parse_args()
+
+    # Setup synthesizer from CLI args if they're specified or no embedded model
+    # is present.
+    if not config or not synthesizer or args.tts_checkpoint or args.tts_config:
         synthesizer = Synthesizer(args)
 
     app.run(debug=config.debug, host='0.0.0.0', port=config.port)
```

**Comment by erogol** [comment]
@reuben you mean discarding my changes any apply the one above?

**Comment by erogol** [comment]
now it works at least with my test.

**Comment by reuben** [comment]
> @reuben you mean discarding my changes any apply the one above?

Yes, the ones to `server.py`. At the very least take out the `parse_args()` call that's outside of the `__main__` block.

**Comment by erogol** [comment]
@reuben could you check if it is now alright?

File: server/server.py [hunk 2]
```diff
@@ -24,23 +24,15 @@ def convert_boolean(x):
...
 embedded_model_folder = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'model')
 checkpoint_file = os.path.join(embedded_model_folder, 'checkpoint.pth.tar')
 config_file = os.path.join(embedded_model_folder, 'config.json')
 
-if os.path.isfile(checkpoint_file) and os.path.isfile(config_file):
-    # Use default config with embedded model files
-    config = create_argparser().parse_args([])
-    config.tts_checkpoint = checkpoint_file
-    config.tts_config = config_file
-    synthesizer = Synthesizer(config)
```
Comment (reuben) [inline server/server.py]: I think removing this block is going to break nginx/uWSGI deployment. I'll check.

**Pull request closed without merging**